### PR TITLE
Add OID Implementation

### PR DESCRIPTION
**Pull request by s00500**
Adds an ObjectIdentifier implementation, stole some of the functions from the go standard library, if you are interested I can also clean it up a bit better :-)

Greetings

### CODE REVIEW

File: ber.go [hunk 1]
```diff
@@ -633,3 +643,166 @@ func NewReal(classType Class, tagType Type, tag Tag, value interface{}, descript
...
 }
+
+func NewOID(classType Class, tagType Type, tag Tag, value interface{}, description string) *Packet {
+	p := Encode(classType, tagType, tag, nil, description)
+
+	switch v := value.(type) {
+	case string:
+		encoded, err := encodeOID(v)
+		if err != nil {
+			fmt.Printf("faled writing %v", err)
```
Comment (johnweldon) [inline ber.go:654]:
```suggestion
			fmt.Printf("failed writing %v", err)
```

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,3 @@
-module github.com/go-asn1-ber/asn1-ber
+module gopkg.in/asn1-ber.v1
```
Comment (johnweldon) [inline go.mod:1]: I would prefer not to change this, at least not in this PR

File: go.mod [hunk 3]
```diff
@@ -1,3 +1,3 @@
...
 
-go 1.13
+go 1.21
```
Comment (johnweldon) [inline go.mod:3]: If there's a way to not have to bump this I'd prefer that.

**Review by johnweldon — CHANGES_REQUESTED** [review]
Thank you for this PR

I like the idea and direction of this.

If you could add some tests and consider the comments I've left we can consider cleaning this up and merging it.

File: go.mod (hunk 3 above)
Comment (s00500) [reply go.mod]: sure, the main reason I did this was to be able to use this in a go.work file

**Comment by s00500** [comment]
Adjusted for comments, added test, ready for re-review :-)

**Comment by johnweldon** [comment]
Thank you @s00500

**Pull request merged**